src/geo/transform.js

```diff
@@ -1306,6 +1310,48 @@ class Transform {
         return this._mercatorZfromZoom(this._minZoomForBounds());
     }
 
+    _calcFogMatrices() {
+        if (!this.fogEnd || !this.fogCulling) {
+            return;
+        }
+
+        const maxVisibleAltitude = this.elevation ? this.elevation.visibleDemTiles.reduce((res, tile) => {
```

Comment:
I think an O(n) tiles loop inside _calcMatrices is a bad idea since it gets called multiple times per frame.. We already have one for `min`. Do you think we can move it to be computed incrementally on tile load and unload?

Reply:
Yep, we can move that in elevation itself and update it once a frame.

Reply:
Simplified and removed this part in https://github.com/mapbox/mapbox-gl-js/commit/e987ba4cb2cac9b50d89a3b5a12d74be3168aa1a and 0555663432f530e7140632e6a51bd9fe6db8d607.